src/cljs/athens/views/blocks/core.cljs

```diff
@@ -217,35 +232,36 @@
            (swap! state assoc :string/previous string :string/local string))
 
          [:div
-          {:class         ["block-container"
-                           (when (and dragging (not is-selected)) "dragging")
-                           (when is-editing "is-editing")
-                           (when is-selected "is-selected")
-                           (when (and (seq children) open) "show-tree-indicator")
-                           (when (and (false? initial-open) (= uid linked-ref-uid)) "is-linked-ref")]
-           :data-uid      uid
-           :on-drag-over  (fn [e] (block-drag-over e block state))
-           :on-drag-leave (fn [e] (block-drag-leave e block state))
-           :on-drop       (fn [e] (block-drop e block state))}
+          {:class          ["block-container"
+                            (when (and dragging (not is-selected)) "dragging")
+                            (when is-editing "is-editing")
+                            (when is-selected "is-selected")
+                            (when (and (seq children) open) "show-tree-indicator")
+                            (when (and (false? initial-open) (= uid linked-ref-uid)) "is-linked-ref")]
+           :data-uid       uid
+           :on-mouse-enter #(swap! state assoc :render-editable true)
+           :on-mouse-leave #(swap! state assoc :render-editable false)
+           :on-click       (fn [e] (doall (.. e stopPropagation) (rf/dispatch [:editing/uid uid])))
+           :on-drag-over   (fn [e] (block-drag-over e block state))
+           :on-drag-leave  (fn [e] (block-drag-leave e block state))
+           :on-drop        (fn [e] (block-drop e block state))}
 
           [presence/presence-popover-info uid {:inline? true}]
 
-          [drop-area-indicator/drop-area-indicator #(when (= drag-target :above) {:opacity "1"})]
+          (when (= (:drag-target @state) :above) [drop-area-indicator/drop-area-indicator {:grid-area "above"}])
 
           [:div.block-body
-           [:button.block-edit-toggle
-            {:on-click (fn [e]
-                         (when (false? (.. e -shiftKey))
-                           (rf/dispatch [:editing/uid uid])))}]
```

Comment:
This button was causing some crazy reagent updates. I replaced it with a conditional `on-click` on the parent.